### PR TITLE
London Class 10 - Rihanna P - JavaScript-Core-1-Coursework-Week2

### PR DESCRIPTION
**Pull request by RihannaP**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name: **Rihanna**
- Your City: **London**
- Your Slack Name: **RihannaP*

# Homework Details

- Module: JS1
- Week: 2

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

### CODE REVIEW

File: mandatory/1-fix-functions.js [hunk 1]
```diff
@@ -10,20 +10,20 @@
 
 */
 
-function getMood() {
+function getMood(answer) {
   let isHappy = true;
```
Comment (VicJerUk) [inline mandatory/1-fix-functions.js]: Good job on this. Just a note... You aren't mutating the value of `isHappy` so `const` would be preferred over `let`

File: mandatory/1-fix-functions.js [hunk 2]
```diff
@@ -10,20 +10,20 @@
...
+  if (isHappy === answer) {
     return "I am happy";
   } else {
     return "I am not happy";
   }
 }
 
 function greaterThan10(num) {
-  let isBigEnough;
+  let isBigEnough = 10;
```
Comment (VicJerUk) [inline mandatory/1-fix-functions.js]: see the previous comment.

File: mandatory/2-function-creation.js [hunk 3]
```diff
@@ -4,7 +4,13 @@
     1. the user should be 18 or older
     2. the user must be logged in
 */
-function isAcceptableUser(userAge, isLoggedIn) {}
+function isAcceptableUser(userAge, isLoggedIn) {
+  standardAge = 18;
+  loggedIn = true;
```
Comment (VicJerUk) [inline mandatory/2-function-creation.js]: Your code is correct and works, but you would want your local variables to be prefixed with a `let` or `const` keyword.

File: mandatory/2-function-creation.js [hunk 4]
```diff
@@ -45,7 +79,12 @@ function canRegister(age) {}
   )
 */
 
-function countReverse(number) {}
+function countReverse(number) {
+  while(number !==0){
+    console.log(number);
+    number--;
+  }
+}
```
Comment (VicJerUk) [inline mandatory/2-function-creation.js]: How will it behave when you give it a negative number? Should we account for that?

File: mandatory/2-function-creation.js [hunk 5]
```diff
@@ -15,27 +21,55 @@ function isAcceptableUser(userAge, isLoggedIn) {}
...
 */
-function buyTwoGetTheCheapestFree(price1, price2) {}
+function buyTwoGetTheCheapestFree(price1, price2) {
+  if (price1 || price2 !== 0){
+    if (price1 < price2){
+      return price2;
+    }
+    else return price1;
+  }
+}
```
Comment (VicJerUk) [inline mandatory/2-function-creation.js]: should we be making sure price1 and price2 are numbers?

**Review by VicJerUk — CHANGES_REQUESTED** [review]
Good effort, have a look at the comments I left you. Let me know if you need them to be clearer.

File: mandatory/2-function-creation.js (hunk 3 above)
Comment (VicJerUk) [reply mandatory/2-function-creation.js]: Sorry should have left you more detailed comment.

**Review by VicJerUk — CHANGES_REQUESTED** [review]
Looks great! There's one comment in the `2-function-creation.js` that haven't been addressed. It's totally my fault I should have written more detailed comment (I've updated it now).

File: mandatory/2-function-creation.js (hunk 3 above)
Comment (RihannaP) [reply mandatory/2-function-creation.js]: Done!

**Pull request closed without merging**